### PR TITLE
Fixes CLI calls for 'port related' operations.

### PR DESCRIPTION
**Pull request by SahilTikale**
Addresses issue #549 
Amended to incorporate re-factored API call that requires to
provide a <switchname> as an argument for every port related operation.

### CODE REVIEW

**Comment by SahilTikale** [comment]
@knikolla @kylehogan 
This is a small PR but import for engage 1.

Please review it when you get time, so that we can merge it with the master to the earliest.

File: tests/unit/obmTest.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	api.node_register('compute-01', obm={
+                  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+                  "host": "ipmihost",
+                  "user": "root",
+                  "password": "tapeworm"})
+
+	node_obj = db.query(model.Node).filter_by(label="compute-01")\
+			.join(model.Obm).join(haas.ext.obm.ipmi.Ipmi).first()
+
+
```
Comment (knikolla) [inline tests/unit/obmTest.py]: Remove extra blank line.

Comment (SahilTikale) [inline tests/unit/obmTest.py]: done.

**Comment by kylehogan** [comment]
can the names of the functions defined in cli.py ('port_register' and 'port_delete') also be changed to match api.py ('switch_register_port' and 'switch_delete_port')?

other than that it looks good

**Comment by SahilTikale** [comment]
> can the names of the functions defined in cli.py ('port_register' and 'port_delete') also be changed to match api.py ('switch_register_port' and 'switch_delete_port')?

I don't think so. Ports are for switches, therefore no point in stating the obvious. 

But I while we are at it, I would like to change the name of `port_delete`
to  `port_unregister`, for consistency sake.

@kylehogan @henn @zenhack 
What do you guys think ?

**Comment by zenhack** [comment]
Re: consistency, nothing else is named `*_unregister`, so I don't see how that helps.

I think @kylehogan is right; we should rename to `switch_*_port`. This is consistent with the `node_*_nic` calls. The naming is a bit odd, and it basically comes down to trying to map a somewhat object-oriented notion onto a purely procedural interface; It's called `node_register_nic` because it's conceptually a method on the node. In the past I've toyed with the idea of doing some meta-programming goop on the server side to make this actually how it works (e.g. `/node/<node>/<stuff>` fetches the node object and then dispatches from there based on `<stuff>`), and while I came to the conclusion that it was a bad idea, it helps illustrate what's going on there conceptually.

I'll point out that the reason for the renaming is *not* that the server and client side should match; the server side is strictly an implementation detail (the public api is the HTTP request) so from the client's perspective it doesn't really matter what they're called. The functions don't even need to map one to one (and they actually don't; see `network_create_simple`). But the above explains the reasons the server-side name was changed, and the same reasoning applies to the client in this case.

File: tests/unit/obmTest.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+"""Unit tests for api.py"""
```
Comment (zenhack) [inline tests/unit/obmTest.py]: This test file has absolutely nothing to do with the content of this patch; it doesn't exercise the CLI at all. Explain?

Comment (SahilTikale) [inline tests/unit/obmTest.py]: Happened by mistake ? Will remove that.

**Comment by SahilTikale** [comment]
So what is the general consensus, change the API names are not ?

**Comment by SahilTikale** [comment]
If we all do not have a consensus, we can just merge this the way it is.
Engage 1 deployment depends on this. 

We will worry about the change of API names in future.

**Comment by zenhack** [comment]
My vote is for making the name change that @kylehogan suggested.

**Comment by SahilTikale** [comment]
This PR is focused on adding cli calls for port operations. So let this do just that.
I will create another PR for changing API names at a later point.

**Comment by knikolla** [comment]
+1

**Comment by zenhack** [comment]
Sounds good to me. Sanity check, since we don't actually have unit tests for the cli: you did try this, correct? If so, you have my +1 as well (and can thus merge).

**Comment by SahilTikale** [comment]
I tried these cli operations before commiting

**Pull request merged**